fhir-search/src/main/java/com/ibm/fhir/search/reference/value/UrlHandlerImpl.java

```diff
@@ -18,11 +18,11 @@
  * http://server/Patient/1
  * -> http://server/Patient/1
  * -> Patient/1
- * -> 1
+ * -> 1 (only when there is one target)
  */
 public class UrlHandlerImpl implements ParameterValueHandler {
 
-    private static final String REGEX = "/[A-z][a-z]{2,10}/[A-Za-z0-9\\-\\.]{1,64}$";
+    private static final String REGEX = "/([A-z][a-z]{2,36}/[A-Za-z0-9\\-\\.]{1,64})$";
```

Comment:
I assume the limit here is for performance reasons?
The longest resource type I found is 30 chars, but I still wonder if we should bump this to like 48 just to be catious.
Alternatively, we could check the actual list of resource names from the resourceTypes valueset and then use the one with max length, but thats probably overkill.

Reply:
we also have a related URL pattern in FHIRUtil.
```
sb.append(")\\/[A-Za-z0-9\\-\\.]{1,64}(\\/_history\\/[A-Za-z0-9\\-\\.]{1,64})?");
```